From 311cfeed93ffae1022d9a3447c7f095744bf8f47 Mon Sep 17 00:00:00 2001
From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Thu, 12 Oct 2017 08:15:17 +0200
Subject: [PATCH] Restore v98 new tab behavior

This patch correcly updates the feed and shows the feed of recently
visited sites, restoring the v98 new tab behavior.

Co-authored-by: uazo <uazo@users.noreply.github.com>
---
 .../chrome/browser/feed/FeedSurfaceMediator.java    | 13 ++++++++++---
 components/ntp_snippets/features.cc                 |  2 +-
 2 files changed, 11 insertions(+), 4 deletions(-)

diff --git a/chrome/android/feed/core/java/src/org/chromium/chrome/browser/feed/FeedSurfaceMediator.java b/chrome/android/feed/core/java/src/org/chromium/chrome/browser/feed/FeedSurfaceMediator.java
index 5abb0dd979033..45887b5a2e73e 100644
--- a/chrome/android/feed/core/java/src/org/chromium/chrome/browser/feed/FeedSurfaceMediator.java
+++ b/chrome/android/feed/core/java/src/org/chromium/chrome/browser/feed/FeedSurfaceMediator.java
@@ -294,9 +294,13 @@ public class FeedSurfaceMediator
     /** Update the content based on supervised user or enterprise policy. */
     void updateContent() {
         mFeedEnabled = FeedFeatures.isFeedEnabled();
-        if (mFeedEnabled == !mTabToStreamMap.isEmpty()) {
-            return;
-        }
+        // the next lines must be removed in bromite/vanadium
+        // since mFeedEnabled is always false and
+        // mTabToStreamMap never empty
+        // addHeaderAndStream is called at least once
+        // if (mFeedEnabled == !mTabToStreamMap.isEmpty()) {
+        //     return;
+        // }
 
         RecyclerView recyclerView = mCoordinator.getRecyclerView();
         if (mSnapScrollHelper != null && recyclerView != null) {
@@ -450,6 +454,9 @@ public class FeedSurfaceMediator
     }
 
     private void addHeaderAndStream(String headerText, Stream stream) {
+        // we need to remove also this when disabled because
+        // the first item inserted is the feed setting
+        if (FeedFeatures.isFeedEnabled() == false) return;
         int tabId = mSectionHeaderModel.get(SectionHeaderListProperties.SECTION_HEADERS_KEY).size();
         mTabToStreamMap.put(tabId, stream);
 
diff --git a/components/ntp_snippets/features.cc b/components/ntp_snippets/features.cc
index 243afdbd79bae..469859bb9991d 100644
--- a/components/ntp_snippets/features.cc
+++ b/components/ntp_snippets/features.cc
@@ -68,7 +68,7 @@ const char kNotificationsDailyLimit[] = "daily_limit";
 const char kNotificationsIgnoredLimitParam[] = "ignored_limit";
 
 const base::Feature kKeepPrefetchedContentSuggestions{
-    "KeepPrefetchedContentSuggestions", base::FEATURE_ENABLED_BY_DEFAULT};
+    "KeepPrefetchedContentSuggestions", base::FEATURE_DISABLED_BY_DEFAULT};
 
 const base::Feature kOptionalImagesEnabledFeature{
     "NTPRemoteSuggestionsOptionalImages", base::FEATURE_ENABLED_BY_DEFAULT};
